context/src/env/parser.rs

```diff
@@ -416,19 +420,22 @@ pub enum BranchClass {
     Merge,
 }
 
-impl TryFrom<(&str, Option<usize>)> for BranchClass {
+impl TryFrom<(&str, Option<usize>, Option<String>)> for BranchClass {
     type Error = CIInfoParseError;
 
-    fn try_from(value: (&str, Option<usize>)) -> Result<Self, Self::Error> {
-        let (branch_name, pr_number) = value;
+    fn try_from(value: (&str, Option<usize>, Option<String>)) -> Result<Self, Self::Error> {
```

Comment:
nit: `Option<String>` feels a little funny since this should only be 1 of `detached`, `merged_result`, or `merge_train`. It might be a bit better for us to "parse" this into an enum and use that here instead.

Reply:
Switched to using an enum and parsing the value